web/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -16,31 +19,52 @@ export class DashboardComponent implements OnInit {
 
   constructor(
     private userService: UserService,
+    private route: ActivatedRoute,
+    private errorService: ErrorService
   ) { }
 
   ngOnInit() {
     this.isResponseLoading = true;
+    this.route.snapshot.paramMap.keys.length ?
+      this.displayPublicDashboard() :
+      this.displayOwnDashboard();
+  }
+
+  displayOwnDashboard() {
     this.userService.getUser()
       .subscribe((user) => {
         if (!user) {
-          console.error('error in fetching user'); // TODO: Handle null user scenario in dashboard UI
+          console.error('Error in fetching user'); // TODO: Handle null user scenario in dashboard UI
           return;
         }
         this.user = { ...user };
-        this.getUserSocialDetails(this.user.uid);
+        this.getUserSocialRecord(this.user.uid);
       });
   }
 
-  getUserSocialDetails(userId: string) {
-    this.userService.getUserSocialDetails(userId)
+  displayPublicDashboard() {
+    const provider = this.route.snapshot.paramMap.get('provider');
+    const name = this.route.snapshot.paramMap.get('name');
+    this.userService.getUserDashboardRecord(provider, name)
+      .pipe(
+        map(value => this.getUserSocialRecord(value.uid)),
+        catchError(error => this.errorService.logError(error))
+      ).subscribe();
```

Comment:
Is something meant to happen here?

Reply:
I see what is done here. On the map, the code handles what needs to happen in the `this.getUserSocialRecord` and essentially the `subscribe` does not need to do anything. If we move, the result of the `this.getUserSocialRecord` to `subscribe`, it would be more cleaner. An abstract example -> 


```this.service.getNumbers().pipe(
    filter(n => n % 5 === 1),
    map(n => n * 2)
  )
  .subscribe(result => console.log(result));```

Reply:
I think this looks better :thumbsup: